storage/height.go

```diff
@@ -0,0 +1,15 @@
+package storage
+
+// HeightIndex defines methods for indexing height.
+// This interface should normally be composed with some other resource we want to index by height.
+type HeightIndex interface {
+	// LatestHeight returns the latest indexed height.
+	LatestHeight() (uint64, error)
+	// FirstHeight at which we started to index. Returns the first indexed height found in the store.
+	FirstHeight() (uint64, error)
+	// SetLatestHeight updates the latest height.
+	// The provided height should either be one higher than the current height or the same to ensure idempotency.
+	// If the height is not within those bounds it will panic!
+	// An error might get returned if there are problems with persisting the height.
```

Comment:
rather than panicking, I think it should return a non-sentinel error indicating the problem. By convention, non-sentinel errors are considered unrecoverable unless specifically stated in the interface.
```suggestion
	// The provided height must either be one higher than the current height or the same to ensure idempotency,
	// otherwise and error is returned. If the height is not within those bounds there is either a bug
	// or state corruption.
	// No errors are expected during normal operation.
```

Reply:
"No errors are expected during normal operation". Are you confused by what it means?

You might find a lot of functions having this same comment in the codebase.

It means, this function would not return any error in normal case, but if it returns, any error it returned must be an exception, which means if the caller get a non-nil error, it should be treated as an exception. However, how an exception should be handled is up to the caller. The caller could decide to panic or "ignore and log", that's the caller's decision, not this function's. 

[A best praice guideline](https://github.com/onflow/flow-go/blob/master/CodingConventions.md#best-practice-guidelines) is to panic when exception happens.